File: problems/2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findMaxK($nums) {
        $arr = [];
        $maxnum = -1;
        foreach($nums as $n){
            if (isset($arr[-$n])){
                if (abs($n) > $maxnum){
                    $maxnum = abs($n);
                }
            }
            $arr[$n]=1;
        }
        return $maxnum;
    }
}

File: problems/2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        uniqueNums = set(nums)
        maxNum = -1
        for n in nums:
            if (n > 0 and -n in uniqueNums):
                if maxNum < n:
                    maxNum = n
        return maxNum